pages/index.js

```diff
@@ -39,14 +42,37 @@ export default function HomePage({
     setModalFilter(!modalFilter);
   }
 
+  function showSearchField() {
+    setIsHidden(!isHidden);
+  }
+
+  const fuse = new Fuse(projects, {
+    keys: ["title", "description", "materials", "steps.desc"],
+    includeScore: true,
+    threshold: 0.3,
+    shouldSort: true,
+    ignoreLocation: true,
+    ignoreFieldNorm: true,
+  });
+  const results = fuse.search(query);
+  console.log("results: ", results);
+  const searchResults = query ? results.map((result) => result.item) : projects;
+
+  function handleSearch(event) {
+    const { value } = event.target;
+    setQuery(value);
```

Comment:
**Debouncing Search Input**: If the project list is large you might want to consider debouncing the search input to reduce the number of searches performed while the user is typing. 

No need to implement it for this project. I just wanted you to be aware of the concept if you hadnt come across it before. 

Here is a [nice visualisation of debouncing](https://kettanaito.com/blog/debounce-vs-throttle).

The idea is that we might not want to perform a search on every keypress. (if it is an expensive operation)

If a user searches for `woodworking`

we might not want to search on every part of the query. 
`w`
`wo`
`woo`
`wood`
`woodw`
etc especially if the user is a fast typer. 

just on the final search term of `woodworking` 

There are a few good articles on debouncing (e.g [here](https://www.dhiwise.com/post/ultimate-guide-to-implementing-react-debounce-effectively))

Reply:
Thanks. That's a very good point. 
I decided to leave this as an idea for later, if maybe we will have a more complex list.